src/cmd/src/options.rs

```diff
@@ -44,6 +47,64 @@ pub struct GreptimeOptions<T> {
     /// The options of each component (like Datanode or Standalone) of GreptimeDB.
     #[serde(flatten)]
     pub component: T,
+
+    /// The options for plugins, keyed by their names. Using this field, we can pass into the
+    /// options for multiple plugins.
+    ///
+    /// For example, suppose we have two plugins' options:
+    ///
+    /// ```rust
+    /// #[derive(Debug, serde::Serialize, serde::Deserialize)]
+    /// struct MyPlugin1Option {
+    ///     a: String,
+    /// }
+    ///
+    /// #[derive(Debug, serde::Serialize, serde::Deserialize)]
+    /// struct MyPlugin2Option {
+    ///     b: Vec<u32>,
+    /// }
+    /// ```
+    ///
+    /// First implementing [`PluginOptions`] for them:
+    ///
+    /// ```no_run
+    /// #[typetag::serde]
+    /// impl PluginOptions for MyPlugin1Option {
+    ///     fn as_any(&self) -> &dyn Any {
+    ///         self
+    ///     }
+    /// }
+    ///
+    /// #[typetag::serde]
+    /// impl PluginOptions for MyPlugin2Option {
+    ///     fn as_any(&self) -> &dyn Any {
+    ///         self
+    ///     }
+    /// }
+    /// ```
+    ///
+    /// Then we put their options in the config file like this:
+    ///
+    /// ```toml
+    /// [plugins.my-plugin1]
+    /// type = "MyPlugin1Option"
+    /// a = "foo"
+    ///
+    /// [plugins.my-plugin2]
+    /// type = "MyPlugin2Option"
+    /// b = [1, 2]
+    /// ```
+    /// (Note that the `type` must equals to the name of the plugin option's definition struct.)
+    ///
+    /// Finally, after loading the GreptimeDB config file, we can `downcast` the
+    /// [`Box<dyn PluginOptions>`] to its real type:
+    ///
+    /// ```no_run
+    /// let options = greptime_options.plugins.remove("my-plugin1").unwrap();
+    /// let plugin1_options: MyPlugin1Option = options.as_any().downcast_ref().unwrap();
+    /// assert_eq(plugin1_options, "foo");
```

Comment:
```suggestion
    /// assert_eq(plugin1_options.a, "foo");
```

Reply:
Can we include a test for this? I'm curious how this works. Judging from the API, it doesn't have a central registry for all types. Maybe typetag maintains a global state internally? But rust doesn't have reflection so how doesn't access those type by, just macros. The macro will no access to the type's fully qualified name...

Reply:
It does have a central registry for all the trait implementations, check this out: https://github.com/dtolnay/typetag?tab=readme-ov-file#so-many-questions

Since there's alreay a plugin option implementation, I decide to discard this one.